control-base/src/main/java/fi/nls/oskari/control/data/MapLayerGroupsHandler.java

```diff
@@ -121,11 +134,39 @@ private void checkForAdminPermission(ActionParameters params) throws ActionExcep
         }
     }
 
-    private void populateFromRequest(ActionParameters params, MaplayerGroup theme) throws ActionException {
-        final Map<String, String> parsed = RequestHelper.parsePrefixedParamsMap(params.getRequest(), PARAM_NAME_PREFIX);
-        for(Map.Entry<String, String> entry : parsed.entrySet()) {
-            // TODO: check against supported locales?
-            theme.setName(entry.getKey(), entry.getValue());
+    protected JSONObject getMapLayerGroup(HttpServletRequest req) throws ActionException {
+        try (InputStream in = req.getInputStream()) {
+            final byte[] json = IOHelper.readBytes(in);
+            final String jsonString = new String(json, StandardCharsets.UTF_8);
+            final JSONObject maplayerGroup = new JSONObject(jsonString);
+            return maplayerGroup;
+        } catch (IOException e) {
+            log.warn(e);
+            throw new ActionException("Failed to read request!");
+        } catch (IllegalArgumentException | JSONException e) {
+            log.warn(e);
+            throw new ActionException("Invalid request!");
+        }
+    }
+    private void populateFromRequest(ActionParameters params, MaplayerGroup maplayerGroup) throws ActionException {
```

Comment:
This function doesn't really need the `ActionParameters params` or `MaplayerGroup maplayerGroup` as parameters. Instead of `ActionParameters params` we only need the  `JSONObject mapLayerGroup`. And instead getting an empty `MaplayerGroup maplayerGroup` as a parameter we could create one inside this function, modify it and then return it.
This would allow the function to focus more on what it's meant to do (transform/map the `JSONObject` the client sent into a `MaplayerGroup` Java object)